engine/src/main/java/org/pentaho/di/trans/DefaultTransManager.java

```diff
@@ -0,0 +1,40 @@
+package org.pentaho.di.trans;
+
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.logging.LogChannelInterface;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class DefaultTransManager implements TransMananger {
+
+  private static final DefaultTransManager instance = new DefaultTransManager();
+
+  Map<String, TransFactory> mapConfigurationTypeTransFactory;
+
+
+  public DefaultTransManager(){
```

Comment:
Sonar will find flag the implementation of this singleton `DefaultTransManager,` in short it needs to be thread safe.

https://rules.sonarsource.com/java/tag/design/RSPEC-6548/